src/getstream/updateBetterChannelMembers.js

```diff
@@ -0,0 +1,176 @@
+        chat_anon_user_info.*
+    FROM users
+    LEFT JOIN chat_anon_user_info
+        ON users.user_id = CAST(chat_anon_user_info.my_anon_user_id as varchar)
+        AND chat_anon_user_info.channel_id = :channelId
+    WHERE users.user_id IN (:userIds)`;
+
+  try {
+    const result = await sequelize.query(query, {
+      replacements: {
+        channelId,
+        userIds
+      },
+      type: Sequelize.QueryTypes.SELECT
+    });
+
+    return result;
+  } catch (e) {
+    console.error('Failed to get chat anonimity user info', e);
+    throw e;
+  }
+};
+
+const __getAnonUserInfoPriority = (memberDataFromDb, member) => {
+  if (memberDataFromDb.anon_user_info_color_code) {
+    return {
+      color_name: memberDataFromDb.anon_user_info_color_name,
+      color_code: memberDataFromDb.anon_user_info_color_code,
+      emoji_name: memberDataFromDb.anon_user_info_emoji_name,
+      emoji_code: memberDataFromDb.anon_user_info_emoji_code
+    };
+  } else if (member.anon_user_info_color_code) {
+    return {
+      color_name: member.anon_user_info_color_name,
+      color_code: member.anon_user_info_color_code,
+      emoji_name: member.anon_user_info_emoji_name,
+      emoji_code: member.anon_user_info_emoji_code
+    };
+  } else {
+    const emoji = BetterSocialConstantListUtils.getRandomEmoji();
+    const color = BetterSocialConstantListUtils.getRandomColor();
+    return {
+      color_name: color.color,
+      color_code: color.code,
+      emoji_name: emoji.name,
+      emoji_code: emoji.emoji
+    };
+  }
+};
+
+const __helperProcessBetterChannelMember = (members, membersDataFromDbMap) => {
+  let newChannelName = '';
+  const better_channel_member = members.map((member) => {
+    const memberDataFromDb = membersDataFromDbMap[member.user_id];
+
+    if (!memberDataFromDb) return member;
+    const {is_anonymous, username, anon_user_info_emoji_name, anon_user_info_color_name} =
+      memberDataFromDb;
+
+    const updatedUsername = is_anonymous
+      ? `${anon_user_info_color_name} ${anon_user_info_emoji_name}`
+      : username;
+
+    newChannelName += `${updatedUsername}${SEPARATOR}`;
+
+    const defaultUser = {
+      ...member,
+      is_anonymous,
+      user: {
+        ...member.user,
+        is_anonymous: is_anonymous,
+        username: updatedUsername,
+        name: updatedUsername
+      }
+    };
+
+    if (is_anonymous) {
+      defaultUser.is_anonymous = is_anonymous;
+      const anon_user_info = __getAnonUserInfoPriority(memberDataFromDb, member);
+      defaultUser.anon_user_info_color_name = anon_user_info.color_name;
+      defaultUser.anon_user_info_color_code = anon_user_info.color_code;
+      defaultUser.anon_user_info_emoji_name = anon_user_info.emoji_name;
+      defaultUser.anon_user_info_emoji_code = anon_user_info.emoji_code;
+    }
+
+    return defaultUser;
+  });
+
+  return {
+    better_channel_member,
+    new_channel_name: newChannelName.slice(0, -SEPARATOR.length)
+  };
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `__helperProcessBetterChannelMember` is doing too much. It's mapping over members, creating a new channel name, and updating each member with new information. Consider breaking this function down into smaller, more manageable functions. This will improve readability and maintainability of your code.



<!-- This is an auto-generated comment by OSS CodeRabbit -->